shared/wallets/wallet/settings-popup/index.js

```diff
@@ -0,0 +1,162 @@
+// @flow
+import * as React from 'react'
+import {
+  Avatar,
+  Box2,
+  ClickableBox,
+  Dropdown,
+  HeaderOnMobile,
+  Icon,
+  MaybePopup,
+  Text,
+} from '../../../common-adapters'
+import {globalColors, globalMargins, globalStyles, platformStyles, styleSheetCreate} from '../../../styles'
+
+export type Props = {
+  name: string,
+  user: string,
+  type: 'default' | 'secondary',
+  currency: string,
+  currencies: Array<string>,
+  onDelete: () => void,
+  onSetDefault: () => void,
+  onEditName: () => void,
+  onCurrencyChange: (currency: string) => void,
+}
+
+const makeDropdownItems = (props: Props) => {
+  let items = [
+    <Box2 centerChildren={true} direction="vertical" key="">
+      <Text type="BodySmall" style={{textAlign: 'center', padding: globalMargins.xsmall}}>
+        Past transactions won’t be affected by this change.
+      </Text>
+    </Box2>,
+  ]
+  for (const s of props.currencies) {
```

Comment:
more idiomatic to have something like
```
items.concat(props.currencies.map(s => makeDropdownItem(s, ...)))
```

Reply:
can you pass other args to the map function? I needed to know which one is selected, so it can be blue.

Reply:
the `s => ...` is a closure (it's just a short form of `(s) => { return ... }`), so it can refer to variables outside its scope, i.e.

```
items.concat(props.currencies.map(s => makeDropdownItem(s,  s === props.currency)))
```
should just work. sorry i wasn't clear!